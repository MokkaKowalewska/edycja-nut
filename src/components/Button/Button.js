import React from "react"
import styled from "styled-components"
import PropTypes from "prop-types"
import { Link } from "gatsby"

const buttonColor = props => (props.white ? "#ffffff" : "#000000")

export const StyledButton = styled(Link)`
  display: block;
  padding: 15px;
  border-radius: 20px;
  background: transparent;
  border: 2px solid;
  border-color: ${buttonColor};
  color: ${buttonColor};
  text-decoration: none;
  transition: all 0.2s ease-in;

  &:hover {
    cursor: pointer;
    background: ${buttonColor};
    color: ${props => (!props.white ? "#ffffff" : "#000000")};
  }
`

const Button = ({ children, ...props }) => (
  <StyledButton {...props}>{children}</StyledButton>
)

Button.propTypes = {
  children: PropTypes.string,
}

export default Button
