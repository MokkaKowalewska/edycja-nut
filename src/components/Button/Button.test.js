import React from "react"
import renderer from "react-test-renderer"

import Button from "./Button"

describe("IndexPage Button", () => {
  it("renders correctly", () => {
    const ButtonTest = renderer
      .create(
        <Button to="/o-mnie" white>
          sprawd≈∫ mnie
        </Button>
      )
      .toJSON()
    expect(ButtonTest).toMatchSnapshot()
  })
})

describe("Kontakt Button", () => {
  it("renders correctly", () => {
    const ButtonTest = renderer
      .create(
        <Button as="a" href="mailto:woj.kowalewski@gmail.com">
          Zapytaj o stawki
        </Button>
      )
      .toJSON()
    expect(ButtonTest).toMatchSnapshot()
  })
})
