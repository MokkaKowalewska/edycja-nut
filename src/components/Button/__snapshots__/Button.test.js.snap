// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<a
  className="sc-AxjAm jWbXFZ"
>
  Zapytaj o wycenę
</a>
`;

exports[`IndexPage Button renders correctly 1`] = `
<a
  className="sc-AxjAm kOxfzR"
  href="/o-mnie"
  white={true}
>
  sprawdź mnie
</a>
`;

exports[`Kontakt Button renders correctly 1`] = `
<a
  className="sc-AxjAm jWbXFZ"
  href="mailto:woj.kowalewski@gmail.com"
>
  Zapytaj o stawki
</a>
`;
